exercises/practice/book-store/.meta/example.cpp

```diff
@@ -0,0 +1,66 @@
+#include <array>
+#include <limits>
+#include <map>
+#include <stdexcept>
+#include <string>
+#include <vector>
+
+#include "book_store.h"
+
+namespace book_store {
+
+static const int PRICE_AFTER_DISCOUNT[6] = {0, 800, 1520, 2160, 2560, 3000};
+
+static int dfs(const std::array<int, 5>& state,
+               std::map<std::array<int, 5>, int>& memo) {
+    auto it = memo.find(state);
+    if (it != memo.end()) return it->second;
+
+    bool empty = true;
+    for (int c : state) {
+        if (c > 0) {
+            empty = false;
+            break;
+        }
+    }
+    if (empty) return memo[state] = 0;
```

Comment:
I'd rather use two statements here:
```suggestion
    if (empty) {
        memo[state] = 0;
        return 0;
    }
```

Reply:
Done.